src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatabilityAnalyzer.Value.cs

```diff
@@ -0,0 +1,244 @@
+
+using System;
+using System.Collections.Immutable;
+using Analyzer.Utilities.PooledObjects;
+using System.Linq;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.GlobalFlowStateAnalysis;
+using Microsoft.CodeAnalysis.FlowAnalysis.DataFlow.ValueContentAnalysis;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis;
+using System.Diagnostics.CodeAnalysis;
+using Analyzer.Utilities;
+
+namespace Microsoft.NetCore.Analyzers.InteropServices
+{
+    using ValueContentAnalysisResult = DataFlowAnalysisResult<ValueContentBlockAnalysisResult, ValueContentAbstractValue>;
+
+    public sealed partial class PlatformCompatabilityAnalyzer
+    {
+        private const string Browser = nameof(Browser);
+        private const string Linux = nameof(Linux);
+        private const string FreeBSD = nameof(FreeBSD);
+        private const string Android = nameof(Android);
+        private const string IOS = nameof(IOS);
+        private const string MacOS = nameof(MacOS);
+        private const string TvOS = nameof(TvOS);
+        private const string WatchOS = nameof(WatchOS);
+        private const string Windows = nameof(Windows);
+
+        private readonly struct RuntimeMethodValue : IAbstractAnalysisValue, IEquatable<RuntimeMethodValue>
+        {
+            private RuntimeMethodValue(string invokedPlatformCheckMethodName, string platformPropertyName, Version version, bool negated)
+            {
+                InvokedMethodName = invokedPlatformCheckMethodName ?? throw new ArgumentNullException(nameof(invokedPlatformCheckMethodName));
+                PlatformName = platformPropertyName ?? throw new ArgumentNullException(nameof(platformPropertyName));
+                Version = version ?? throw new ArgumentNullException(nameof(version));
+                Negated = negated;
+            }
+
+            public string InvokedMethodName { get; }
+            public string PlatformName { get; }
+            public Version Version { get; }
+            public bool Negated { get; }
+
+            public IAbstractAnalysisValue GetNegatedValue()
+                => new RuntimeMethodValue(InvokedMethodName, PlatformName, Version, !Negated);
+
+            public static bool TryDecode(
+                IMethodSymbol invokedPlatformCheckMethod,
+                ImmutableArray<IArgumentOperation> arguments,
+                ValueContentAnalysisResult? valueContentAnalysisResult,
+                INamedTypeSymbol osPlatformType,
+                [NotNullWhen(returnValue: true)] out RuntimeMethodValue? info)
+            {
+                // Accelerators like OperatingSystem.IsPlatformName()
+                if (arguments.IsEmpty)
+                {
+                    var platformName = SwitchPlatformName(invokedPlatformCheckMethod.Name);
```

Comment:
Nit: This file uses all TryGet patterns for parsing APIs, may want to do the same here for consistency.